presentation/src/main/java/me/androidbox/presentation/event/viewmodel/EventViewModel.kt

```diff
@@ -133,6 +140,72 @@ class EventViewModel @Inject constructor(
                     )
                 }
             }
+            is EventScreenEvent.OnVisitorEmailChanged -> {
+                _eventScreenState.update { eventScreenState ->
+                    eventScreenState.copy(
+                        visitorEmail = eventScreenEvent.visitorEmail
+                    )
+                }
+            }
+            is EventScreenEvent.OnShowVisitorDialog -> {
+                _eventScreenState.update { eventScreenState ->
+                    eventScreenState.copy(
+                        shouldShowVisitorDialog = eventScreenEvent.shouldShowVisitorDialog,
+                        isEmailVerifiedSuccess = true,
+                        visitorEmail = ""
+                    )
+                }
+            }
+            is EventScreenEvent.CheckVisitorExists -> {
+                verifyVisitorEmail(eventScreenEvent.visitorEmail)
+            }
+        }
+    }
+
+    private fun verifyVisitorEmail(visitorEmail: String) {
+        viewModelScope.launch {
+            val responseState = verifyVisitorEmailUseCase.execute(visitorEmail)
+
+            when(responseState) {
+                is ResponseState.Loading -> {
+                    /* TODO Show loading */
+                }
+                is ResponseState.Success -> {
+                    responseState.data?.let { _attendee ->
+                        val attendee = _attendee.copy(
+                            email = _attendee.email,
+                            fullName = _attendee.fullName,
+                            userId = _attendee.userId,
+                            remindAt = _attendee.remindAt,
+                            eventId = eventScreenState.value.eventId,
+                            isGoing = _attendee.isGoing
+                        )
+
+                        _eventScreenState.update { eventScreenState ->
+                            eventScreenState.copy(
+                                isEmailVerifiedSuccess = true,
+                                listOfAttendee = eventScreenState.listOfAttendee + attendee
+                            )
+                        }
+
+                        /* TODO Add this attendee to the event and update this event in db */
```

Comment:
When we get a valid response back with an visitor.
1) This attendee will be added to the Event, and that Event will be updated in the room DB with the added attendee?
2) We have call our work manager to upload that event as well. As we did for the insertion of a new event?

Reply:
Yes, after an attendee is added via the dialog, you just add it to your local event state of your ViewModel. Once you hit save, you upload the event with all the added attendees. 
I'd recommend to use WorkManager to upload an event as well, since it might contain up to 10 photos which could take a bit